### ISSUE TITLE
Rails 5: previous_changes includes [nil, nil] entry for postrgesql json column

### ISSUE THREAD

**Issue by nbrustein** [rails/rails#27201]
### Steps to reproduce

 * Create an active record model using the postgresql adapter that has a column of type `json`
 * Load an instance of the model from the database where the value of the column is `NULL`
 * Save the instance

### Expected behavior
The column should not be included in previous_changes, since it did not change

### Actual behavior
`previous_changes` will be `{"my_column" => [nil, nil]}`

### System configuration
**Rails version**:
Rails 5.0.0.1

**Ruby version**:
ruby 2.3.0p0 (2015-12-25 revision 53290) [x86_64-darwin15]

**Notes**:

This only happens in Rails 5.  In rails 4, previous_changes would have been empty, as expected.

Here is a test that passes in rails 4 but fails in rails 5, assuming you have a `users` table with a `provider_user_info` column of type `json`

````
describe ActiveRecord::Base do

    describe "previous_changes" do

        it "should not include a json field changing from nil to nil" do
            user = User.where(provider_user_info: nil).first
            user.save!

            expect(user.previous_changes['provider_user_info']).to be_nil
        end

    end

end
````

**Comment by booleanbetrayal** [comment]
Just did a bit of a dive ... I'm seeing the `previous_changes` spec failing in `5.0.0.1`, but both appear to be passing in the `5-0-stable` branch.

**Comment by nbrustein** [comment]
Even though the test passes in `5-0-stable`, I'm still seeing the same issue in our actual code when using `5-0-stable`.  So, there is some way to get into the state where previous_changes shows `[nil, nil]`, but the test included above is no longer sufficient to produce the issue.  (I've looked at this together with @booleanbetrayal and confirmed we're both seeing the same thing)

**Comment by sgrif** [comment]
This was fixed in 21675fdc, which was backported to 5-0-stable in b601c8c3. Neither commit is in a released version of Rails. I suspect you're seeing an issue in your actual code because the value was serialized as JSON null, and is going to be changed to SQL NULL when re-saved. You can update your entire database with `UPDATE whatever SET that_col = NULL WHERE that_col = 'null';`. There is no deprecation/migration strategy, as Active Record was unable to query for fields with the old value.

If you still think there's a bug, please comment with a script that can be used to reproduce the issue using our [standard bug report template](https://github.com/rails/rails/blob/master/guides/bug_report_templates/active_record_master.rb) and I will re-open this issue.

**Comment by sgrif** [comment]
Actually I lied. The fix is included in https://rubygems.org/gems/rails/versions/5.0.1.rc1, which should fix your issue.
